internal/machineconfigpools/machineconfigpools_test.go

```diff
@@ -0,0 +1,197 @@
+/*
```

Comment:
likely in this file, but not completely sure where this should be, shouldn't we have tests to cover multiple kubeletconfigs that the first one in the list is not matching the nodegroups configured in nrop resource, and the second one does?

Reply:
good idea, I'll work on that